### PR TITLE
feat: return validity page

### PR DESCRIPTION
**Pull request by dannydaviesIW**
# Description

-   This PR adds a new page into the return ticket user journey flow which collects information about the validity period of the return ticket.

# Testing instructions

-   Pull down this branch and run the site locally. Validate that the new page appears as expected and behaves as expected.

# Type of change

-   [X] feat - A new feature
-   [ ] fix - A bug fix
-   [ ] docs - Documentation only changes
-   [ ] codestyle - Changes that do not affect the meaning of the code (white-space, formatting, missing semi-colons, etc)
-   [ ] refactor - A code change that neither fixes a bug nor adds a feature
-   [ ] perf - A code change that improves performance
-   [ ] test - Adding missing tests or correcting existing tests
-   [ ] build - Changes that affect the build system or external dependencies (example scopes: gulp, broccoli, npm)
-   [ ] ci - Changes to our CI configuration files and scripts (example scopes: Travis, Circle, BrowserStack, SauceLabs)
-   [ ] chore - Other changes that don't modify src or test files
-   [ ] revert - Reverts a previous commit
-   [ ] content - Changes to content or copy

# Checklist:

-   [X] Able to run pr locally
-   [X] Followed acceptance criteria
-   [X] My code follows the style guidelines of this project
-   [X] I have performed a self-review of my own code
-   [ ] I have made corresponding changes to the documentation if applicable
-   [X] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

File: pull_request_template.md [hunk 1]
```diff
@@ -24,7 +24,6 @@
 # Checklist:
 
 -   [ ] Able to run pr locally
--   [ ] Lighthouse performed (Accessibility 90+ minimum)
```
Comment (tejpowar) [inline pull_request_template.md:27]: how comes we have removed this in the template?

Comment (RobCatton) [reply pull_request_template.md:27]: Because we no longer value what it tells us (DAC audit stuff).

File: src/pages/api/returnValidity.ts [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+                    .number()
+                    .typeError(inputError)
+                    .when('duration', {
+                        is: 'day',
+                        then: yup
+                            .number()
+                            .typeError(inputError)
+                            .integer(inputError)
+                            .min(1, daysAmountError)
+                            .max(31, daysAmountError),
```
Comment (RobCatton) [inline src/pages/api/returnValidity.ts:41]: I dont think we want to put caps on what they can input

File: src/pages/api/returnValidity.ts [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+import { ErrorInfo, NextApiRequestWithSession } from '../../interfaces';
+import { updateSessionAttribute } from '../../utils/sessions';
+import { RETURN_VALIDITY_ATTRIBUTE } from '../../constants';
+
+const radioButtonError = 'Choose one of the options below';
+const inputError = 'Enter a whole number';
+const daysAmountError = 'Enter a number of days between 1 and 31';
+const weeksAmountError = 'Enter a number of weeks between 1 and 52';
+const monthsAmountError = 'Enter a number of months between 1 and 72';
+const yearsAmountError = 'Enter a number of years greater than 1';
```
Comment (RobCatton) [inline src/pages/api/returnValidity.ts:14]: Should this say greater than zero?

File: src/pages/api/returnValidity.ts [hunk 4]
```diff
@@ -0,0 +1,133 @@
...
+import { redirectToError, redirectTo } from './apiUtils/index';
+import { isSessionValid } from './apiUtils/validator';
+import { ErrorInfo, NextApiRequestWithSession } from '../../interfaces';
+import { updateSessionAttribute } from '../../utils/sessions';
+import { RETURN_VALIDITY_ATTRIBUTE } from '../../constants';
+
+const radioButtonError = 'Choose one of the options below';
+const inputError = 'Enter a whole number';
+const daysAmountError = 'Enter a number of days between 1 and 31';
+const weeksAmountError = 'Enter a number of weeks between 1 and 52';
```
Comment (RobCatton) [inline src/pages/api/returnValidity.ts:12]: As below, not sure we have a reason to be capping this, or at least not this low.

File: src/pages/api/returnValidity.ts [hunk 5]
```diff
@@ -0,0 +1,133 @@
...
+const monthsAmountError = 'Enter a number of months between 1 and 72';
+const yearsAmountError = 'Enter a number of years greater than 1';
+const selectError = 'Choose one of the options from the dropdown list';
+
+export const returnValiditySchema = yup
```
Comment (RobCatton) [inline src/pages/api/returnValidity.ts:17]: Upon testing this, we error if the user inputs a valid number, and follows it with a space. Should we trim the string to remove leading and trailing whitespace?

**Review by RobCatton — APPROVED** [review]

**Review by RobCatton — APPROVED** [review]

**Pull request merged**